### ISSUE TITLE
HYPRE_v2.29.0's HYPRE_IJMatrixInitialize starts to return hypre_error_flag=1

### ISSUE THREAD

**Issue by t34ma** [hypre-space/hypre#946]
Dear Developers,

I tried to use the hypre v2.29.0 for my fortran FEM program.
Then the subroutine HYPRE_IJMatrixInitialize starts to return hypre_error_flag=1.
The calculation itself continues successfully and it seems to get right solution.
I did not make any change to my code. Just change the library from v2.28.0.

Testsuya Mishima

**Comment by victorapm** [comment]
Hi Testsuya,

would you be able to recompile hypre with the configure option `--with-print-errors` and let me know if you get an error message for the issue you mentioned?

**Comment by t34ma** [comment]
Hi Victor A. P. Magri,

 

Thank you for your reply.

 

I solved it myself.

Although I have been a hypre user for a long time, I didn’t know HYPRE_INIT/FINALIZE 

was added (since the middle of 2018 as far as I checked), because my application worked 

fine without calling it. 

I checked what happened using debugger step by step. Then I noticed the HYPRE_IJMatrixInitialize 

generates warning because hypre is not yet initialized. So I added HYPRE_init/HYPRE_finalize 

to my codes. Then it works fine. I guess HYPRE-2.29.0 slightly changed the behavior of initialize.

 

Regards,

Tetsuya Mishima
